invisible_cities/cities/detsim_waveforms.py

```diff
@@ -0,0 +1,53 @@
+import numpy as np
+
+##################################
+######### WAVEFORMS ##############
+##################################
+def bincounter(xs : np.array,
+               dx : float =1.,
+               x0 : float =0.):
+    ixs = ((xs - x0) // dx).astype(int)
+    return np.unique(ixs, return_counts=True)
+
+
+def create_waveform(times : np.array,
+                    pes   : np.array,
+                    bins  : np.array,
+                    wf_bin_time : float,
+                    nsamples    : int) -> np.array:
+    wf = np.zeros(len(bins))
+
+    if np.sum(pes)==0:
+        return wf
+    if nsamples==0:
```

Comment:
The code outside the `if` is generic enough to handle `nsamples = 1`, which is what you are dealing with in this `if` statement, right? Maybe you should forbid `nsamples=0` and if the input is `1` you know it works (a test to demonstrate it would be nice)

Reply:
As it is now, `nsamples=1` redistribute the counts in the same bin, which is equivalent to do not redistribute at all. In this case the function runs the count distribution loop. 

In order to afford this and improve performance (to do not enter in the loop), I suggest to keep the conditional for `nsamples=0` and `nsamples=1` such as: `if nsamples in [0, 1]` or equivalent.

Reply:
Unless there is a performance impact as you claim, I don't see a good reason to do that. It adds a bit of noise and, more importantly, it duplicates the code, which is not good. In any case, we can leave the time profiling to the end when the code is more mature and stable.

In any case, having who values in which `nsamples` implies a different behaviour is a bit fishy. Stick to one, which I would say is `nsamples = 1`. And unless there is a relevant performance difference, don't make it a special case at all!